src/sonic_ax_impl/mibs/vendor/cisco/ciscoPfcExtMIB.py

```diff
@@ -0,0 +1,263 @@
+            return None
+
+    def cpfcIfRequests(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the counter for the respective sub_id/table.
+        """
+        oid = self.get_oid(sub_id)
+        if oid is None:
+            return None
+
+        counter_name = 'SAI_PORT_STAT_PFC_3_RX_PKTS'
+
+        if oid in self.oid_lag_name_map:
+            counter_value = 0
+            for lag_member in self.lag_name_if_name_map[self.oid_lag_name_map[oid]]:
+                counter_value += self._get_counter(mibs.get_index(lag_member), counter_name)
+
+            # truncate to 32-bit counter
+            return counter_value & 0x00000000ffffffff
+        else:
+            return self._get_counter(oid, counter_name)
+
+
+    def cpfcIfIndications(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the counter for the respective sub_id/table.
+        """
+        oid = self.get_oid(sub_id)
+        if oid is None:
+            return None
+
+        counter_name = 'SAI_PORT_STAT_PFC_3_TX_PKTS'
+
+        if oid in self.oid_lag_name_map:
+            counter_value = 0
+            for lag_member in self.lag_name_if_name_map[self.oid_lag_name_map[oid]]:
+                counter_value += self._get_counter(mibs.get_index(lag_member), counter_name)
+
+            # truncate to 32-bit counter
+            return counter_value & 0x00000000ffffffff
+        else:
+            return self._get_counter(oid, counter_name)
+
+
+class PfcPrioUpdater(PfcUpdater):
+    def __init__(self):
+        super().__init__()
+        self.max_prio = 8
+
+    def queue_index(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the 0-based interface ID.
+        """
+        if len(sub_id) >= 2:
+            return sub_id[1] - 1
+        return None
+
+    def get_next(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the next sub id.
+        """
+        try:
+            if sub_id is None:
+               return (self.if_range[0][0], 1)
+
+            if len(sub_id) < 2:
+               return (sub_id[0], 1)
+
+            if sub_id[1] >= self.max_prio:
+                return None
+
+            right = sub_id[1] + 1
+
+            return (sub_id[0], right)
+        except Exception as e:
+            mibs.logger.error("failed to get next oid with error = {}".format(str(e)))
+
+    def requestsPerPriority(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the counter for the respective sub_id/table.
+        """        
+        port_oid = ''
+        queue_index = ''
+        try:
+            port_oid = self.get_oid((sub_id[0],))
+            queue_index = self.queue_index(sub_id)
+            if port_oid is None or queue_index is None:
+                raise IndexError()
```

Comment:
>raise IndexError() [](start = 16, length = 18)

Is it only to print a warning log? If yes, just call logger function directly.

Reply:
We could have a exception in the line 187 and need to verify port_oid and queue_index. I gather the log message in one place instead of duplication.

Reply:
Replaced raise exception to return none